aten/src/ATen/cpu/vec256/vec256_complex_double.h

```diff
@@ -218,15 +218,7 @@ template <> class Vec256<c10::complex<double>> {
     AT_ERROR("not supported for complex numbers");
   }
   Vec256<c10::complex<double>> exp() const {
-    //exp(a + bi)
-    // = exp(a)*(cos(b) + sin(b)i)
-    auto exp = Sleef_expd4_u10(values);                               //exp(a)           exp(b)
-    exp = _mm256_blend_pd(exp, _mm256_permute_pd(exp, 0x05), 0x0A);   //exp(a)           exp(a)
-
-    auto sin_cos = Sleef_sincosd4_u10(values);                        //[sin(a), cos(a)] [sin(b), cos(b)]
-    auto cos_sin = _mm256_blend_pd(_mm256_permute_pd(sin_cos.y, 0x05),
-                                   sin_cos.x, 0x0A);                  //cos(b)           sin(b)
-    return _mm256_mul_pd(exp, cos_sin);
+    return map(std::exp);
```

Comment:
So I appreciate this approach and understand we took it on the last PR, but there we now have to reimplement something performant. Would it be easier to fix these vectorized impls instead of first removing and then re-adding them?

Reply:
I tried fixing it to some extent to match the standard
```text
Errors are reported consistent with math_errhandling

If the implementation supports IEEE floating-point arithmetic,

std::exp(std::conj(z)) == std::conj(std::exp(z))
If z is (±0,+0), the result is (1,+0)
If z is (x,+∞) (for any finite x), the result is (NaN,NaN) and FE_INVALID is raised.
If z is (x,NaN) (for any finite x), the result is (NaN,NaN) and FE_INVALID may be raised.
If z is (+∞,+0), the result is (+∞,+0)
If z is (-∞,y) (for any finite y), the result is +0cis(y)
If z is (+∞,y) (for any finite nonzero y), the result is +∞cis(y)
If z is (-∞,+∞), the result is (±0,±0) (signs are unspecified)
If z is (+∞,+∞), the result is (±∞,NaN) and FE_INVALID is raised (the sign of the real part is unspecified)
If z is (-∞,NaN), the result is (±0,±0) (signs are unspecified)
If z is (+∞,NaN), the result is (±∞,NaN) (the sign of the real part is unspecified)
If z is (NaN,+0), the result is (NaN,+0)
If z is (NaN,y) (for any nonzero y), the result is (NaN,NaN) and FE_INVALID may be raised
If z is (NaN,NaN), the result is (NaN,NaN)
where cis(y) is cos(y) + i sin(y)
```

But couldn't handle all the case. So for starters I thought having the correct code makes sense.

Also from the CI failures, I think compilers on Windows and Mac probably have different implementation.

Reply:
After seeing the latest PR with its greatly extended vectorized code I'm inclined to agree with you that supporting a vectorized code path may not make sense. Let's discuss this below.